src/libcore/iter/adapters/zip.rs

```diff
@@ -62,6 +62,17 @@ impl<A, B> DoubleEndedIterator for Zip<A, B> where
     fn next_back(&mut self) -> Option<(A::Item, B::Item)> {
         ZipImpl::next_back(self)
     }
+
+    #[inline]
+    fn nth_back(&mut self, mut n: usize) -> Option<(A::Item, B::Item)> {
+        while let Some(x) = ZipImpl::next_back(self) {
```

Comment:
The default implementation of `nth_back` calls `Iterator::next_back` in a loop, which seems like it'd be just as good as this one?  It looks like `ZipImpl::nth` exists, so I'd have expected the change here to add `ZipImpl::nth_back` and specialize it the same way.